### PR TITLE
fix(SF-1157): Update state on update

### PR DESCRIPTION
**Pull request by Vic-Dev**

### CODE REVIEW

File: test/unit/refinement-crumbs.ts [hunk 1]
```diff
@@ -70,7 +73,7 @@ suite('RefinementCrumbs', ({ expect, spy, stub, itShouldProvideAlias }) => {
 
     it('should update field', () => {
       refinementCrumbs.previousField = 'brand';
-      refinementCrumbs.flux = { off: () => null, on: () => null } as any;
+      refinementCrumbs.flux = <any>{ off: () => null, on: () => null };
```
Comment (jmbtutor) [inline test/unit/refinement-crumbs.ts:76]: Interestingly, the preferred way to cast is using `as`. I'm not asking you to revert this, though.

File: src/refinement-crumbs/index.ts [hunk 2]
```diff
@@ -6,19 +6,17 @@ class RefinementCrumbs {
   previousField: string;
 
   init() {
-    let selectedRefinementsUpdated;
-    let navigationSelector;
     switch (this.props.storeSection) {
       case StoreSections.PAST_PURCHASES:
-        selectedRefinementsUpdated = Events.PAST_PURCHASE_SELECTED_REFINEMENTS_UPDATED;
-        navigationSelector = (field) => this.select(Selectors.pastPurchaseNavigation, field);
+        this.state.selectedRefinementsUpdated = Events.PAST_PURCHASE_SELECTED_REFINEMENTS_UPDATED;
```
Comment (jmbtutor) [inline src/refinement-crumbs/index.ts:11]: As long as we're sure that `state` is an object at this point, this is fine. I think that's the case.

**Review by jmbtutor — APPROVED** [review]
Looks reasonable.

File: src/refinement-crumbs/index.ts (hunk 2 above)
Comment (Vic-Dev) [reply src/refinement-crumbs/index.ts:11]: yes https://github.com/groupby/storefront-core/blob/master/src/core/tag/index.ts#L22

**Pull request merged**